Week3/homework/Masoud/creditCardValidator/creditCardValidator.js

```diff
@@ -0,0 +1,100 @@
+'use strict'
+
+const validateCreditNumber = (creditCardNumber) => {
+
+  const numStringLength = creditCardNumber.length;
+
+  //Input must be 16 characters
+  function checkLength(creditCardNumber) {
+    if (numStringLength !== 16){
+      return `Invalid! The input ${creditCardNumber} should contains 16 characters!`;
+    } else {
+      return 'success!';
+    }
+  }
+
+  //All characters must be numbers
+  function checkType(creditCardNumber) {
+    const index = [];
+
+    for (let i = 0 ; i < numStringLength ; i++) {
+      if (creditCardNumber[i] == 0 || creditCardNumber[i] == 1 || creditCardNumber[i] == 2 || creditCardNumber[i] == 3 || creditCardNumber[i] == 4 || creditCardNumber[i] == 5 || creditCardNumber[i] == 6 || creditCardNumber[i] == 7 || creditCardNumber[i] == 8 || creditCardNumber[i] == 9) {
+        
+      } else {
+        index.push('NaN');
+      }
+    }
+    
+    if (index.length !== 0){
+      return `Invalid! The input ${creditCardNumber} should contains only numbers!`;
+    } else {
+      return 'success!';
+    }
+
+  }
+  
+  //At least two different numbers should be represented
+  function atLeastTwo(creditCardNumber) {
+    const index = [];
+
+    for (let i = 1 ; i < numStringLength - 1 ; i++) {
```

Comment:
just use the `length` here. Make your functions as independent as possible.  You have `creditCardNumber` use that to get the length